@here/harp-map-controls/lib/CameraKeyTrackAnimation.ts

```diff
@@ -0,0 +1,216 @@
+
+import { CameraAnimation } from "./MapAnimations";
+import { CameraPose, CameraPoseOptions } from "./Utils";
+
+const logger = LoggerManager.instance.create("CameraKeyTrackAnimation");
+
+export interface ControlPointOptions extends CameraPoseOptions {
+    timestamp: number;
+}
+/**
+ * A ControlPoint used to defined the Track of the [[CameraKeyTrackAnimation]]
+ */
+export class ControlPoint extends CameraPose {
+    timestamp: number = 0;
+    constructor(options: ControlPointOptions) {
+        super(options);
+        this.timestamp = options.timestamp;
+    }
+}
+
+export interface CameraKeyTrackAnimationOptions {
+    controlPoints: ControlPoint[];
+    repeat?: boolean;
+    name?: string;
+    interpolation?: THREE.InterpolationModes;
+    loop?: THREE.AnimationActionLoopStyles; //types seem outdate too, should be THREE.LoopModes
+    rotateOnlyClockwise?: boolean;
+    repetitions?: number;
+}
+
+class AnimationDummy extends THREE.Object3D {
+    distance: number = 0;
+
+    constructor(name: string) {
+        super();
+        this.name = name;
+    }
+}
+
+/**
+ * A [[CameraAnimation]] along of a set of [[ControlPoint]]s
+ */
+export class CameraKeyTrackAnimation extends CameraAnimation {
+    private m_animationClip: THREE.AnimationClip;
+    private m_animationMixer: THREE.AnimationMixer;
+    private m_animationAction: THREE.AnimationAction;
+    private m_dummy: AnimationDummy = new AnimationDummy("dummy");
+    private m_azimuthAxis = new THREE.Vector3(0, 0, 1);
+    private m_altitudeAxis = new THREE.Vector3(1, 0, 0);
+
+    private m_startTime: number = 0;
+    private m_animateCb: () => void;
+    private m_stopCb: () => void;
+
+    constructor(private m_mapView: MapView, private m_options: CameraKeyTrackAnimationOptions) {
+        super(m_mapView, m_options.name || "flyoveranimation");
+
+        const interpolation =
+            this.m_options.interpolation !== undefined
+                ? this.m_options.interpolation
+                : THREE.InterpolateSmooth;
+
+        const loop = this.m_options.loop !== undefined ? this.m_options.loop : THREE.LoopOnce;
+        const repetitions =
+            this.m_options.repetitions !== undefined ? this.m_options.repetitions : 1;
+
+        this.m_options.rotateOnlyClockwise =
+            this.m_options.rotateOnlyClockwise !== undefined
+                ? this.m_options.rotateOnlyClockwise
+                : true; //default to true
+
+        const timestamps = this.m_options.controlPoints.map(point => {
+            return point.timestamp;
+        });
+        const posValues: [number?] = [];
+        const rotValues: [number?] = [];
+        this.m_options.controlPoints.map(point => {
+            const worldPos = this.m_mapView.projection.projectPoint(point.geoCoordinate);
+            posValues.push(worldPos.x);
+            posValues.push(worldPos.y);
+            posValues.push(worldPos.z);
+
+            const rot = new THREE.Quaternion();
```

Comment:
I have strong feeling that we don't need quaternions here, but only properly interpolate `LookAtParams.heading` and `LookAtParams.tilt` between particular animation frames. Only suggestion, ...

Reply:
I have not tried, but I was hoping for a more constant animation speed and avoidance of gimbal lock this way